html/user/per_app_list.php

```diff
@@ -96,34 +107,72 @@ function show_row($item, $apps, $is_team, $i) {
     echo "</tr>\n";
 }
 
-function show_list($is_team, $appid, $is_total) {
-    $x = $is_team?"teams":"participants";
-    page_head("Top $x by application");
-    $apps = BoincApp::enum("deprecated=0");
-    if (!$appid) {
-        $appid = $apps[0]->id;
-    }
-    start_table('table-striped');
-    show_header($is_team, $apps, $appid, $is_total);
+function get_top_items($is_team, $appid, $is_total, $offset) {
+    global $items_per_page;
     $x = $is_total?"total":"expavg";
     if ($is_team) {
-        $items = BoincCreditTeam::enum("appid=$appid order by $x desc");
+        return  BoincCreditTeam::enum("appid=$appid order by $x desc limit $offset,$items_per_page");
     } else {
-        $items = BoincCreditUser::enum("appid=$appid order by $x desc");
+        return  BoincCreditUser::enum("appid=$appid order by $x desc limit $offset,$items_per_page");
     }
-    $i = 0;
-    foreach ($items as $item) {
-        show_row($item, $apps, $is_team, $i);
-        $i++;
-    }
-    end_table();
-    page_tail();
 }
 
 $is_team = get_int('is_team', true);
 $appid = get_int('appid', true);
 $is_total = get_int('is_total', true);
+$items_per_page = 20;
+$offset = get_int('offset', true);
+if (!$offset) $offset=0;
+if ($offset % $items_per_page) $offset = 0;
+$x = $is_team?"teams":"participants";
+page_head("Top $x by application");
```

Comment:
Can you make this translatable? Something like `tra("Top %1 by application", $x)` should work.